sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/aad/implementation/conditions/AllWebApplicationConditionTests.java

```diff
@@ -0,0 +1,85 @@
+            .withUserConfiguration(WebApplicationConditionConfig.class)
+            .run(assertConditionMatch(false));
+    }
+
+    @Test
+    void testWebAppConditionWhenNoOAuth2ResourceDependency() {
+        this.contextRunner
+            .withPropertyValues("spring.cloud.azure.active-directory.credential.client-id = fake-client-id")
+            .withClassLoader(new FilteredClassLoader(BearerTokenAuthenticationToken.class))
+            .withUserConfiguration(WebApplicationConditionConfig.class)
+            .run(assertConditionMatch(true));
+    }
+
+    @Test
+    void testWebAppConditionWhenNoOAuth2ClientDependency() {
+        this.contextRunner
+            .withPropertyValues("spring.cloud.azure.active-directory.credential.client-id = fake-client-id")
+            .withClassLoader(new FilteredClassLoader(ClientRegistration.class))
+            .withUserConfiguration(WebApplicationConditionConfig.class)
+            .run(assertConditionMatch(false));
+    }
+
+    @Test
+    void testWebAppConditionWhenApplicationTypeIsWebApplication() {
+        this.contextRunner
+            .withPropertyValues(
+                "spring.cloud.azure.active-directory.credential.client-id = fake-client-id",
+                "spring.cloud.azure.active-directory.application-type=web_application")
+            .withUserConfiguration(WebApplicationConditionConfig.class)
+            .run(assertConditionMatch(true));
+    }
+
+    @Test
+    void testWebAppConditionWhenApplicationTypeIsResourceServer() {
+        this.contextRunner
+            .withPropertyValues(
+                "spring.cloud.azure.active-directory.credential.client-id = fake-client-id",
+                "spring.cloud.azure.active-directory.application-type=resource_server")
+            .withUserConfiguration(WebApplicationConditionConfig.class)
+            .run(assertConditionMatch(false));
+    }
+
+    @Test
+    void testWebAppConditionWhenApplicationTypeIsResourceServerWithOBO() {
+        this.contextRunner
+            .withPropertyValues(
+                "spring.cloud.azure.active-directory.credential.client-id = fake-client-id",
+                "spring.cloud.azure.active-directory.application-type=resource_server_with_obo")
+            .withUserConfiguration(WebApplicationConditionConfig.class)
+            .run(assertConditionMatch(false));
+    }
+
+    @Test
+    void testWebAppConditionWhenApplicationTypeIsWebApplicationAndResourceServer() {
+        this.contextRunner
+            .withPropertyValues(
+                "spring.cloud.azure.active-directory.credential.client-id = fake-client-id",
+                "spring.cloud.azure.active-directory.application-type=web_application_and_resource_server")
+            .withUserConfiguration(WebApplicationConditionConfig.class)
+            .run(assertConditionMatch(true));
+    }
+
+    @Configuration
+    @Conditional(AllWebApplicationCondition.class)
+    static class WebApplicationConditionConfig extends Config { }
```

Comment:
Keep it align to condition class name. Here is should be `AllWebApplicationConditionConfig`. 
No need to change it if you rename `AllWebApplicationConditionConfig` to `WebApplicationCondition`.